src/dataviews/dataviews-factory.js

```diff
@@ -26,62 +26,55 @@ module.exports = Model.extend({
     this._analysisCollection = opts.analysisCollection;
   },
 
-  createCategoryModel: function (layerModel, attrs) {
+  createCategoryModel: function (attrs) {
     _checkProperties(attrs, ['column']);
-    attrs = this._generateAttrsForDataview(layerModel, attrs, CategoryDataviewModel.ATTRS_NAMES);
+    attrs = this._generateAttrsForDataview(attrs, CategoryDataviewModel.ATTRS_NAMES);
     attrs.aggregation = attrs.aggregation || 'count';
     attrs.aggregation_column = attrs.aggregation_column || attrs.column;
 
-    var categoryFilter = new CategoryFilter({
-      layer: layerModel
-    });
+    var categoryFilter = new CategoryFilter();
 
     return this._newModel(
       new CategoryDataviewModel(attrs, {
         map: this._map,
         vis: this._vis,
-        layer: layerModel,
```

Comment:
We can also get rid of the `analysisCollection` option (and probably `vis` and `map` too) \o/.

Also, this factory probably doesn't need `analysisCollection` as a dependency.

Reply:
`map` is used in dataviews for:
- Respond to changes like `view_bounds_ne`, `center`, `zoom`.
- Get the view bound.

`vis` is used in dataviews for:
- Reload the map (do an instantiation)

So i'm not sure we can eliminate those two properties unless you have something in mind that I'm not aware of, like `Engine` being responsible for triggering those changes to dataviews.

Reply:
You're totally right! I misunderstood the diff. We can only get rid of `analysisCollection`.